pkg/ccl/testccl/sqlccl/sqlccl_test: TestShowTransferState failed

pkg/ccl/testccl/sqlccl/sqlccl_test.TestShowTransferState [failed](https://mesolite.cluster.engflow.com/invocations/default/4adc367d-de04-48f6-982f-a16f4e3808bd?testReportRun=1&testReportShard=13&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvc3FsY2NsOnNxbGNjbF90ZXN0) on master @ [78fa126dfc749e620c28531c1659e35bf98034ec](https://github.com/cockroachdb/cockroach/commits/78fa126dfc749e620c28531c1659e35bf98034ec):


```
=== RUN   TestShowTransferState
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestShowTransferState246603600
    test_log_scope.go:81: use -show-logs to present logs inline
    show_transfer_state_test.go:268: -- test log scope end --
test logs left over in: outputs.zip/logTestShowTransferState246603600
--- FAIL: TestShowTransferState (2.05s)
=== RUN   TestShowTransferState/errors
    --- FAIL: TestShowTransferState/errors (0.02s)
=== RUN   TestShowTransferState/errors/root_user
    show_transfer_state_test.go:219: 
        	Error Trace:	pkg/ccl/testccl/sqlccl/sqlccl_test/pkg/ccl/testccl/sqlccl/show_transfer_state_test.go:219
        	Error:      	Not equal: 
        	            	expected: "cannot create token for root user"
        	            	actual  : "session revival tokens are not supported on this cluster"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-cannot create token for root user
        	            	+session revival tokens are not supported on this cluster
        	Test:       	TestShowTransferState/errors/root_user
        --- FAIL: TestShowTransferState/errors/root_user (0.00s)
=== RUN   TestShowTransferState/successful_transfer
    show_transfer_state_test.go:163: 
        	Error Trace:	pkg/ccl/testccl/sqlccl/sqlccl_test/pkg/ccl/testccl/sqlccl/show_transfer_state_test.go:163
        	Error:      	Received unexpected error:
        	            	failed to connect to ``host=127.0.0.1 user=testuser database=``: server error (ERROR: session revival tokens are not supported on this cluster (SQLSTATE 28000))
        	Test:       	TestShowTransferState/successful_transfer
    --- FAIL: TestShowTransferState/successful_transfer (0.01s)
=== RUN   TestShowTransferState/with_transfer_key
    show_transfer_state_test.go:142: 
        	Error Trace:	pkg/ccl/testccl/sqlccl/sqlccl_test/pkg/ccl/testccl/sqlccl/show_transfer_state_test.go:142
        	Error:      	Should be false
        	Test:       	TestShowTransferState/with_transfer_key
    --- FAIL: TestShowTransferState/with_transfer_key (0.20s)
=== RUN   TestShowTransferState/without_transfer_key
    show_transfer_state_test.go:78: 
        	Error Trace:	pkg/ccl/testccl/sqlccl/sqlccl_test/pkg/ccl/testccl/sqlccl/show_transfer_state_test.go:78
        	Error:      	Should be false
        	Test:       	TestShowTransferState/without_transfer_key
        	Messages:   	expected null error, got session revival tokens are not supported on this cluster
    --- FAIL: TestShowTransferState/without_transfer_key (0.02s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=13</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowTransferState.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/ccl/testccl/sqlccl/sqlccl_test.TestShowTransferState [failed](https://tanzanite.cluster.engflow.com/invocations/default/8ff3a3d0-5634-4327-b1ec-ed00720f22b6?testReportRun=8&testReportShard=13&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvc3FsY2NsOnNxbGNjbF90ZXN0) on master @ [9e48858c6c8b22af4ec1159bcff6e233e7bfddff](https://github.com/cockroachdb/cockroach/commits/9e48858c6c8b22af4ec1159bcff6e233e7bfddff):


```
=== RUN   TestShowTransferState
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestShowTransferState136549065
    test_log_scope.go:81: use -show-logs to present logs inline
    show_transfer_state_test.go:268: -- test log scope end --
test logs left over in: outputs.zip/logTestShowTransferState136549065
--- FAIL: TestShowTransferState (2.16s)
=== RUN   TestShowTransferState/without_transfer_key
    show_transfer_state_test.go:78: 
        	Error Trace:	pkg/ccl/testccl/sqlccl/sqlccl_test/pkg/ccl/testccl/sqlccl/show_transfer_state_test.go:78
        	Error:      	Should be false
        	Test:       	TestShowTransferState/without_transfer_key
        	Messages:   	expected null error, got session revival tokens are not supported on this cluster
    --- FAIL: TestShowTransferState/without_transfer_key (0.02s)
=== RUN   TestShowTransferState/errors
    --- FAIL: TestShowTransferState/errors (0.02s)
=== RUN   TestShowTransferState/errors/root_user
    show_transfer_state_test.go:219: 
        	Error Trace:	pkg/ccl/testccl/sqlccl/sqlccl_test/pkg/ccl/testccl/sqlccl/show_transfer_state_test.go:219
        	Error:      	Not equal: 
        	            	expected: "cannot create token for root user"
        	            	actual  : "session revival tokens are not supported on this cluster"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1 +1 @@
        	            	-cannot create token for root user
        	            	+session revival tokens are not supported on this cluster
        	Test:       	TestShowTransferState/errors/root_user
        --- FAIL: TestShowTransferState/errors/root_user (0.00s)
=== RUN   TestShowTransferState/successful_transfer
    show_transfer_state_test.go:163: 
        	Error Trace:	pkg/ccl/testccl/sqlccl/sqlccl_test/pkg/ccl/testccl/sqlccl/show_transfer_state_test.go:163
        	Error:      	Received unexpected error:
        	            	failed to connect to ``host=127.0.0.1 user=testuser database=``: server error (ERROR: session revival tokens are not supported on this cluster (SQLSTATE 28000))
        	Test:       	TestShowTransferState/successful_transfer
    --- FAIL: TestShowTransferState/successful_transfer (0.01s)
=== RUN   TestShowTransferState/with_transfer_key
    show_transfer_state_test.go:142: 
        	Error Trace:	pkg/ccl/testccl/sqlccl/sqlccl_test/pkg/ccl/testccl/sqlccl/show_transfer_state_test.go:142
        	Error:      	Should be false
        	Test:       	TestShowTransferState/with_transfer_key
    --- FAIL: TestShowTransferState/with_transfer_key (0.20s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=8</code>
 - <code>shard=13</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestShowTransferState.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/ccl/testccl/sqlccl/show_transfer_state_test.go
+++ b/pkg/ccl/testccl/sqlccl/show_transfer_state_test.go
@@ -14,8 +14,8 @@ import (
 	"github.com/cockroachdb/cockroach-go/v2/crdb"
 	"github.com/cockroachdb/cockroach/pkg/base"
 	"github.com/cockroachdb/cockroach/pkg/security/username"
+	"github.com/cockroachdb/cockroach/pkg/testutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
@@ -26,7 +26,6 @@ import (
 func TestShowTransferState(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-	skip.WithIssue(t, 128125)
 
 	ctx := context.Background()
 	s, mainDB, _ := serverutils.StartServer(t, base.TestServerArgs{
@@ -211,9 +210,11 @@ func TestShowTransferState(t *testing.T) {
 		t.Run("root_user", func(t *testing.T) {
 			var key string
 			var errVal, sessionState, sessionRevivalToken gosql.NullString
-			err := tenantDB.QueryRow(`SHOW TRANSFER STATE WITH 'bar'`).Scan(&errVal, &sessionState, &sessionRevivalToken, &key)
-			require.NoError(t, err)
-
+			testutils.SucceedsSoon(t, func() error {
+				// Waiting for the cluster setting to propagate to the tenant.
+				return tenantDB.QueryRow(`SHOW TRANSFER STATE WITH 'bar'`).
+					Scan(&errVal, &sessionState, &sessionRevivalToken, &key)
+			})
 			require.True(t, errVal.Valid)
 			require.Equal(t, "cannot create token for root user", errVal.String)
 			require.False(t, sessionState.Valid)

